test/contrib/automotive/obd/scanner.uts

```diff
@@ -140,111 +135,111 @@ from subprocess import call
+     ECUResponse(session=range(0, 255), security_level=0, responses=s6_mid00),
+     ECUResponse(session=range(0, 255), security_level=0, responses=s8_tid00),
+     ECUResponse(session=range(0, 255), security_level=0, responses=s9_iid00),
+     ECUResponse(session=range(0, 255), security_level=0, responses=s1_pid01),
+     ECUResponse(session=range(0, 255), security_level=0, responses=s1_pid03),
+     ECUResponse(session=range(0, 255), security_level=0, responses=s1_pid0B),
+     ECUResponse(session=range(0, 255), security_level=0, responses=s1_pid0F)]
 
-+ Simulate scanner
 
-= Create responder
-exit_if_no_isotp_module()
-# Ensures the responder is running before sending the first request
-ready = Event()
-
-class MockResponder(Thread):
-    daemon = True
-    def __init__(self):
-        Thread.__init__(self)
-        self._stopped = Event()
-    def process_request(self, req, sock):
-        if req.service == 1:
-            resp = s1_queue.get()
-            sock.send(resp)
-        elif req.service == 3:
-            resp = s3_queue.get()
-            sock.send(resp)
-    def run(self):
-        with ISOTPSocket(new_can_socket(iface0), 0x7e8, 0x7e0, basecls=OBD, padding=True) as sock:
-            while not self._stopped.is_set() or not (s1_queue.empty() and s3_queue.empty()):
-                sock.sniff(timeout=5, started_callback=ready.set, store=False, prn=lambda p: self.process_request(p, sock))
-    def stop(self):
-        self._stopped.set()
-
-responder = MockResponder()
-responder.start()
-
-= Get ids
-exit_if_no_isotp_module()
-drain_bus(iface0)
-with ISOTPSocket(new_can_socket(iface0), 0x7e0, 0x7e8, basecls=OBD, padding=True) as socket:
-    ready.wait(timeout=5)
-    all_ids_set = set(range(1, 256))
-    supported_ids = _supported_id_numbers(socket, 0.1, OBD_S01, 'pid', False)
-    unsupported_ids = all_ids_set - supported_ids
++ Simulate scanner
 
 = Run scanner
+
 exit_if_no_isotp_module()
+
 drain_bus(iface0)
-# timeout to avoid a deadlock if the test which sets this event fails
-with ISOTPSocket(new_can_socket(iface0), 0x7e0, 0x7e8, basecls=OBD, padding=True) as socket:
-    ready.wait(timeout=5)
-    data = obd_scan(socket, 0.1, True, True, verbose=False)
-    dtc = data[0]
-    supported = data[1]
-    unsupported = data[2]
-
-= Cleanup
-exit_if_no_isotp_module()
-responder.stop()
+
+with ISOTPSocket("vcan0", 0x7e8, 0x7e0, basecls=OBD, padding=True) as ecu:
+    answering_machine = ECU_am(supported_responses=example_responses, main_socket=ecu, basecls=OBD)
+    sim = threading.Thread(target=answering_machine, kwargs={'count': 12})
+    sim.start()
+    try:
+        with ISOTPSocket(new_can_socket(iface0), 0x7e0, 0x7e8, basecls=OBD, padding=True) as socket:
+            all_ids_set = set(range(1, 256))
+            supported_ids = _supported_id_numbers(socket, 0.1, OBD_S01, 'pid', False)
+            unsupported_ids = all_ids_set - supported_ids
+
```

Comment:
I had problems with empty lines on indented code blocks in UTScapy, in the past. Just as remark.